ui/src/shared/components/FieldList.js

```diff
@@ -58,6 +64,49 @@ class FieldList extends Component {
     this.props.onFill(fill)
   }
 
+  handleToggleField = field => {
+    const {
+      query,
+      onToggleField,
+      addInitialField,
+      initialGroupByTime: time,
+      isKapacitorRule,
+    } = this.props
+    const {fields, groupBy} = query
+    const initialGroupBy = {...groupBy, time}
+
+    if (!_.size(fields)) {
```

Comment:
why _.size and not .length

Reply:
also this could be a compound condition:

```javascript
if ( fields.length || isKapacitorRule ) {
  onToggleField(field)
}

addInitialField(field, initialGroupBy)
```

Reply:
_.size(undefined) => 0
undefined.length => runtime error